### PR TITLE
Fix bugs in NuE/NueScatterGen_module.cc

### PR DESCRIPTION
**Pull request by tomjunk**
The random sampling in TF1::GetRandom was hanging.  Turns out setting Npx helps.  See 

https://root-forum.cern.ch/t/problem-generating-a-distribution-using-tf1-getrandom-function/44812

for a discussion.  Instead of computing Npx, it is just a fcl parameter.

I also deleted the old version in NuEScattering -- it wasn't being built anyway and had the same name as the active code.

I also set fFlatESpectrum from the fcl parameter -- was forgotten before.

### CODE REVIEW

File: dunesim/EventGenerator/NuE/NuEScatterGen_module.cc [hunk 1]
```diff
@@ -217,8 +234,17 @@ void evgen::NuEScatterGen::reconfigure(fhicl::ParameterSet const & p)
...
+  fFlatESpectrum = p.get<bool>("FlatESpectrum",false);
+  
   if (!fFlatESpectrum || !fCustomNuFlavBR) {
     fEventRateFileName = p.get<std::string>("EventRateFileName");
+    fNpxNueE           = p.get<int>("fNpxNueE",1500);
+    fNpxNumuE          = p.get<int>("fNpxNumuE",1500);
+    fNpxNutauE         = p.get<int>("fNpxNutauE",1500);
+    fNpxNuebarE        = p.get<int>("fNpxNuebarE",1500);
+    fNpxNumubarE       = p.get<int>("fNpxNumubarE",1500);
+    fNpxNutaubarE      = p.get<int>("fNpxNutaubarE",1500);
```
Comment (absolution1) [inline dunesim/EventGenerator/NuE/NuEScatterGen_module.cc:246]: The default values could be dropped as they could end up being a point of user error (e.g. if someone typos the name of the variable in their own fcl)

**Review by absolution1 — COMMENTED** [review]
Hi Tom.
Thanks for adding these.
I added a single comment about the default values when reading in the FCL pset.

Cheers!

**Comment by mhlswang** [comment]
Hi Tom, I tried running it and found the following minor typos in the lines that Dom referred to above:
the "f" in the  "fNpxNu*" in p.get<int>("fNpxNu*",1500) needs to be removed.  Other than this, everything looks ok.

**Review by mhlswang — CHANGES_REQUESTED** [review]
Hi Tom, I tried running it and found the following minor typos in the lines that Dom referred to above:
the `f` in the `fNpxNu*` in `p.get<int>("fNpxNu*")` needs to be removed. Other than this, everything looks ok.

**Comment by tomjunk** [comment]
Thanks!  I just pushed a commit that hopefully addresses both of these comments.

**Review by absolution1 — APPROVED** [review]
Thanks for making the change @tomjunk 
This looks good to me!

**Pull request merged**